libyul/optimiser/SimplificationRules.cpp

```diff
@@ -46,7 +46,7 @@ SimplificationRule<yul::Pattern> const* SimplificationRules::findFirstMatch(
 	if (!instruction)
 		return nullptr;
 
-	static SimplificationRules rules;
+	static SimplificationRules rules(_dialect);
```

Comment:
This way, the rules list will always use the dialect it was first initialized with. You need a static (maybe LazyInit) variable of type `std::map<EVMVersion, SimplificationRules>`.

Reply:
Ah ok.. I thought that the compiler will only be initialized with exactly one `EVMVersion`..

Reply:
Never be sure about that :)